app/services/solidus_bolt/payments/capture_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module Payments
+    class CaptureService < SolidusBolt::BaseService
+      attr_reader :payment, :order_amount, :capture_amount
+
+      def initialize(payment:, order_amount:, capture_amount:)
+        @payment = payment
+        @order_amount = order_amount
+        @capture_amount = capture_amount
+        super
+      end
+
+      def call
+        return if payment.completed?
+
+        if order_amount == capture_amount
+          payment.complete!
+        else
+          Spree::PaymentCaptureEvent.create(payment: payment, amount: Money.new(capture_amount).to_d)
```

Comment:
the core gist of the method is creating a [capture_event for the amount captured](https://github.com/spree/spree/blob/ab448523bdc112306b67e73bcf16ef132e791df7/core/app/models/spree/payment/processing.rb#L45), which is what I'm doing in the partial capture

the part above (lines 39-43) call the Gateway to make a capture request, which is what we want to avoid since the payment was already captured on Bolt
```
    24: def capture(float_amount, response_code, gateway_options)
    25:   binding.pry
 => 26:   payment = gateway_options[:originator]
```

Reply:
Ok, now i see your point. But there is also a method called 'split_uncaptured_amount' that should executed anyway. In that way the payment amount will be updated and a new payment with the remaining amount will be created

Reply:
Interesting. I based my code on the [snippet from Solidus](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment/processing.rb#L46-L67) instead of the one from Spree that you are sharing, and Solidus doesn't have that `split_uncaptured_amount` method